### PR TITLE
NetworkDevice (NAD):  Allow SNMP parameters

### PR DESCRIPTION
**Pull request by staley1975**
Allows adding network devices (nads) with SNMP parameters, and setting SNMP parameters to existing network devices (nads)

### CODE REVIEW

**Comment by alagoutte** [comment]
Hi @staley1975 

Thanks for your contribution, it is possible to add some test (on Tests/integration/NetworkDevice.Tests.ps1) ?

File: PowerArubaCP/Public/NetworkDevice.ps1 [hunk 1]
```diff
@@ -297,6 +321,10 @@ function Set-ArubaCPNetworkDevice {
         [Parameter (Mandatory = $false)]
         [string]$radius_secret,
         [Parameter (Mandatory = $false)]
+        [string]$snmp_version,
```
Comment (alagoutte) [inline PowerArubaCP/Public/NetworkDevice.ps1:324]: Use ValidateSet v1/v2c ?

File: PowerArubaCP/Public/NetworkDevice.ps1 [hunk 2]
```diff
@@ -345,6 +373,16 @@ function Set-ArubaCPNetworkDevice {
             $_nad | add-member -name "radius_secret" -membertype NoteProperty -Value $radius_secret
         }
 
+        if ($PsBoundParameters.ContainsKey('snmp_version')) {
```
Comment (alagoutte) [inline PowerArubaCP/Public/NetworkDevice.ps1:376]: if community_string is empty ?

File: PowerArubaCP/Public/NetworkDevice.ps1 [hunk 3]
```diff
@@ -345,6 +373,16 @@ function Set-ArubaCPNetworkDevice {
...
+            $snmp_read = @{
+            snmp_version = $snmp_version
```
Comment (alagoutte) [inline PowerArubaCP/Public/NetworkDevice.ps1:378]: add 4space (for align)

```
$snmp_read = @{
    snmp_version = $snmp_version
    community_string....
```

**Review by alagoutte — CHANGES_REQUESTED** [review]

**Comment by alagoutte** [comment]
can you also look for test ?

may be also move example to the end (and use SW6 ?)

**Comment by alagoutte** [comment]
Thanks, i will try and merge if it is ok

**Comment by staley1975** [comment]
> Thanks, i will try and merge if it is ok

Sounds good, and thank you for the feedback.

**Comment by alagoutte** [comment]
> > Thanks, i will try and merge if it is ok
> 
> Sounds good, and thank you for the feedback.

Tested with 6.11 and 6.12 ! Working !

Thanks for your contribution, it will publish a new release on some next week...

**Pull request merged**